### PR TITLE
[VIT-2947] Adopt Android SDK 1.0.0-beta.14

### PR DESCRIPTION
**Pull request by andersio**
* Adopt the new Vital Health Connect permission API from Android.

* Clean up the usage of CoroutineScope. Only one is needed, and it only needs to be cancelled in `cleanUp()` or when we recreate `VitalHealthConnectManager` in `configure()`.

### CODE REVIEW

File: packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt [hunk 1]
```diff
@@ -3,23 +3,20 @@ package com.vitalhealthreactnative
...
-import io.tryvital.vitalhealthconnect.model.SyncStatus
-import io.tryvital.vitalhealthconnect.model.VitalResource
-import io.tryvital.vitalhealthconnect.model.WritableVitalResource
+import io.tryvital.vitalhealthconnect.model.*
 import kotlinx.coroutines.*
 import java.time.Instant
-import kotlin.reflect.KClass
+
+const val VITAL_HEALTH_ERROR = "VitalHealthError"
+const val VITAL_REQUEST_CODE = 1984
```
Comment (RuiAAPeres) [inline packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt:19]: What's the goal of this value?

Comment (andersio) [reply packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt:19]: It is the way of Android deeplinking to other apps or system dialogs.

The app initiates the call with an `Intent` and a "request code". Then when the system delivers the callback via `onActivityResult`, the app is provided with "request code" for it to differentiate the calls it has made.

Acts similarly to the context pointer in, well...., good old Objective-C KVO.

Comment (andersio) [reply packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt:19]: https://github.com/tryVital/vital-react-native/blob/d4f80d6d46e1d3938a12a0e7a1f5317070c62dde/packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt#L164-L167

https://github.com/tryVital/vital-react-native/blob/d4f80d6d46e1d3938a12a0e7a1f5317070c62dde/packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt#L326-L330

Comment (RuiAAPeres) [reply packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt:21]: Is this a George Orwell reference then?

**Review by RuiAAPeres — APPROVED** [review]

File: packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt (hunk 1 above)
Comment (andersio) [reply packages/vital-health-react-native/android/src/main/java/com/vitalhealthreactnative/VitalHealthReactNativeModule.kt:21]: yes

**Pull request merged**